### PR TITLE
Topic 3 services

### PR DESCRIPTION
**Pull request by cristian-rodrig**
Topic 3 services

### CODE REVIEW

File: src/main/java/com/bootcamp/topic3/cart/CartItem.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class CartItem {
+	public static int idGenerator = 0;
+	public int id;
+	public String name;
+	public double price;
+	public int quantity;
+
+	public CartItem(String name, double price, int quantitie) {
+
+		this.id = idGenerator++;
```
Comment (carlos-lopez-glb) [inline src/main/java/com/bootcamp/topic3/cart/CartItem.java]: To create an idGenerator I suggest to use [UUID](https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html). Your approcach is not thread safety and less reliable.

`this.id = UUID.randomUUID().toString();`

File: src/main/java/com/bootcamp/topic3/service/ShoppingCartServiceImp.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	
+	private ArrayList<CartItem> items;
+	
+	
+	public ShoppingCartServiceImp() {
+		items = new ArrayList<CartItem>();
+	}
+
+	/**
+	 * agrega un item al carro
```
Comment (carlos-lopez-glb) [inline src/main/java/com/bootcamp/topic3/service/ShoppingCartServiceImp.java]: The javadoc must be in english

File: src/main/java/com/bootcamp/topic3/service/ShoppingCartServiceImp.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	 */
+	public void addCartItem(CartItem item) {
+		items.add(item);
+	}
+
+	/**
+	 * borra un item del carro
```
Comment (carlos-lopez-glb) [inline src/main/java/com/bootcamp/topic3/service/ShoppingCartServiceImp.java]: The javadoc should be on the interface, the javadoc in the implemetation will be copied from there.

> The interface should have all the information about the contract, basically what the method does, the description of parameters, return values and so on.

> Unless there's some extra information that isn't clear from the interface description (there rarely is), the implementation documentation should then simply link to the interface method.

Also the implementation's javadoc can be override
https://dzone.com/articles/inheriting-javadoc-method-comments

File: src/test/java/com/bootcamp/topic3/test/ShoppingCartServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		service.addCartItem(c1);
+		service.addCartItem(c2);
+	}
+	
+	@After
+	public void after(){
+		service.clearCart();
+	}
+
+	@Test
```
Comment (carlos-lopez-glb) [inline src/test/java/com/bootcamp/topic3/test/ShoppingCartServiceTest.java:36]: This Unit Test suite includes only possitive tests, what about the negative test? I mean, the cases of add and delete a null object is not covered.

Comment (cristian-rodrig) [reply src/test/java/com/bootcamp/topic3/test/ShoppingCartServiceTest.java:36]: Hi I had some problems to understand how pull request works . Finally I did the corrections and created a diferent PR  https://github.com/cristian-rodrig/gitBootCamp/pull/4

**Pull request merged**